### ISSUE TITLE
Could you give me a hint?

### ISSUE THREAD

**Issue by newforrestgump001** [dlib-users/dnn#1]
I have samples composed of A,B,C, which are 3 classes. And train classifier using the 3 classes. If I give the classifier the D sample, As far as I know, it will one of the labels, A,B or C. But for D sample, A B or C are all wrong labels. How can I know it is not A,B or C (it is abnormal or novelty.)  Thanks  a lot!

**Comment by arrufat** [comment]
I am assuming you are training with `loss_multiclass_log`, right? That loss function uses a softmax, which means, the sum of the three possible outputs has to add to 1 always. So, it can output something like 0.33 for each class when it's not sure, however, if it's slightly more sure that it's one class, it might get overconfident on that class.

For that case, I like to train classifiers with the [`loss_multibinary_log`](http://dlib.net/dlib/dnn/loss_abstract.h.html#loss_multibinary_log_). Instead of a softmax, it uses a sigmoid for each output independently, so any combination can be possible, even outputting everything to 0 and everything to 1 (that's why it's sometimes called multi label).

You can check an example on how to use it in the [tests](https://github.com/davisking/dlib/blob/adca7472dfa2ed2df735dcc341fefb4cbe747f2d/dlib/test/dnn.cpp#L3636-L3697).

**Comment by newforrestgump001** [comment]
It is very helpful, and thanks a lot! BTW, Are you familiar with Anomaly Detection for defect inspection which is unpredictable, it may be everything you can imagine.
